docs-internal/config-metadata.md

```diff
@@ -0,0 +1,404 @@
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+                <forceJavacCompilerUse>true</forceJavacCompilerUse>
+                <annotationProcessorPaths>
+                    <path>
+                        <groupId>io.helidon.config</groupId>
+                        <artifactId>helidon-config-metadata-processor</artifactId>
+                        <version>${helidon.version}</version>
+                    </path>
+                </annotationProcessorPaths>
+            </configuration>
+        </plugin>
+    </plugins>
+</build>
+```
+3. Update the `module-info.java` by adding
+```java
+requires static io.helidon.config.metadata;
+```
+4. Annotate the configured class using `@Configured` - usually the builder class. If there is only a factory method, annotate
+   the class containing the factory method
+5. Annotate builder methods using `@ConfiguredOption` - the type of the parameter will be used as type of the property, provides
+   full customization using annotation properties
+6. In case a factory method is the only one available, annotate it with repeating `@ConfiguredOption` to list all annotations
+7. Look at existing examples if in doubt
+8. Check the output in `target/classes/META-INF/helidon` to see what was generated
+
+## Output file format
+
+The file is `META-INF/helidon/config-metadata.json`
+
+
+### Root Element
+Root of the file is an array of module objects
+
+```json
+[
+     {
+          "module": ""
+     }
+]
+```
+
+### Module element
+Module is equivalent to `module-info.java`. This approach allows merging of multiple metadata files into a single
+file.
+
+```json
+{
+"module": "module name (from module-info.java)",
+"types" : []
+}
+```
+
+### Type element
+Each type represents a configurable unit.
+
+"type":
+```json
+{
+     "type": "fully qualified type name of the configured class (the class built by builder, or created by factory method)",
+     "standalone": true,
+     "prefix": "server",
+     "description": "Documentation of this type",
+     "producers": ["methods"],
+     "inherits": ["fully qualified type names of superclasses/interfaces this type extends/implements"],
+     "provides": ["fully qualified type names of provided services"],
+     "options": []
+}
+```
+
+prefix - configuration prefix, if standalone, this is the root configuration key (such as `server`, `tracing`),
+if nested, this is a prefix added before all options of this type (such as each Security provider)
+standalone - if set to true, this type is configurable in the root of configuration (such as webserver, security, tracing, metrics),
+otherwise it is a nested type in another configuration (such as WebServerTls, OidcConfig)
+default is `false` (if not defined in document)
+producers (methods in general) - format is "fully qualified type name#method name (method parameter types)", example:
+`io.helidon.security.providers.common.OutboundTarget.Builder#build()`
```

Comment:
This renders poorly. The lines all run together.

Reply:
added bullet points